ui/src/components/dialog.tsx

```diff
@@ -0,0 +1,70 @@
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import TextField from "@mui/material/TextField";
+import { ChangeEvent, ReactNode, useState } from "react";
+
+type NewDialogProps = {
+  title: string;
+  titleId: string;
+  textLabel: string;
+  className: string;
+  callback: (name: string) => void;
+};
+
+export function useDialog(props: NewDialogProps): [ReactNode, () => void] {
+  const [open, setOpen] = useState(false);
+  const show = () => {
+    setOpen(true);
+  };
+
+  const [name, setName] = useState(props.title);
+  const onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+  };
+
+  const onCreate = () => {
```

Comment:
I also proposed it should be property for the hint at beginning. But then, I think the text input type button hint will mostly be "create/confirm".  largely possibility not change.
Quick question: Do you think the rename dialog should be "confirm" and the dataset.tsx dialog should be "create"?

Reply:
Given that it's a generic dialog the functions and variables should have generic names rather than create or rename that are specific to certain usages. I'm not sure what you mean about the button mostly being create/confirm. That's the two use cases we have but there isn't a reason to limit it to those options.

I would use "Rename" and "Create" for the buttons. Confirm is a more generic name to use in the code.

Reply:
I understand your idea and agree with you. Thanks for letting me know the graininess for a property.

Reply:
It doesn't look like Create was renamed to Confirm.

Reply:
Oh. I think i might misunderstand the comment. I change the buttonhint as a property then it can transfer, and we can set the button shows whatever we want. like create ,or  Confirm. Because this is a generic module. I just use the onCreate as the function name. Do you mean I should change the name from "onCreate" to "onConfirm" ?

Reply:
Maybe we can call it as onClick? It will be more generic.